pandas/io/stata.py

```diff
@@ -65,6 +67,9 @@
 
 from pandas.io.common import get_handle
 
+if TYPE_CHECKING:
```

Comment:
Do you need a check that the Python is 3.8 or later here, with an import from typing_extensions if 3.7?  You can do it like it is done everywhere else, so if this is standard then this is ok (means can't type check on 3.7 unless changed). For an example that supports 3.7,

https://github.com/numpy/numpy/blob/bfb1ac86588456a78af63fa314b768f4ebc764ff/numpy/random/_generator.pyi#L47

Reply:
you can just import this from pandas._typing (so when we change to 3.8 this would be fine)